### PR TITLE
refactor(popover): use core tokens

### PR DESCRIPTION
**Pull request by lunarfusion**
## Description
This updates the Popover component to use core tokens

Uses beta release. Hold for full release.

## Related issue(s)

-

## How has this been tested?

-   [ ] _Test case 1_
    1. Go here
    2. Do this
-   [ ] _Test case 2_
    1. Go here
    2. Do this

## Types of changes

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to change)
-   [ ] Chore (minor updates related to the tooling or maintenance of the repository, does not impact compiled assets)

## Checklist

-   [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
-   [ ] My code follows the code style of this project.
-   [ ] If my change required a change to the documentation, I have updated the documentation in this pull request.
-   [ ] I have read the **[CONTRIBUTING](<(https://github.com/adobe/spectrum-web-components/blob/main/CONTRIBUTING.md)>)** document.
-   [ ] I have added tests to cover my changes.
-   [ ] All new and existing tests passed.
-   [ ] I have reviewed at the Accessibility Practices for this feature, see: [Aria Practices](https://www.w3.org/TR/wai-aria-practices/)

## Best practices

This repository uses conventional commit syntax for each commit message; note that the GitHub UI does not use this by default so be cautious when accepting suggested changes. Avoid the "Update branch" button on the pull request and opt instead for rebasing your branch against `main`.

### CODE REVIEW

**Comment by lunarfusion** [comment]
WIP Assessment of VRT Failures

# Failure Pattern 1: Expected due to design change
Extra vertical spacing is due to the addition of vertical padding using token popover-top-to-content-area.
Screen Shot 2022-11-08 at 4 56 54 PM

**Comment by pfulton** [comment]
This has been merged in the CSS repo and the release has been published: `@spectrum-css/popover@6.0.0`

**Comment by Westbrook** [comment]
Question on "Failure Pattern 5: Expected":

It seems that the Popover is adding this `popover-top-to-content-area` padding, which is the same amount of external space as `Menu` applies via margin. Are these measures meant to be leveraged together, or should they both be expected to collapse into each other in some way?

**Comment by hunterloftis** [comment]
Just making sure that these diffs are expected:

https://6fb3035dee9ee944fc231e1d21676146--spectrum-web-components.netlify.app/review/#OverlayStories/customizedClickContent.png:
image

https://6fb3035dee9ee944fc231e1d21676146--spectrum-web-components.netlify.app/review/#OverlayStories/definedOverlayElement.png:
image

File: packages/popover/src/Popover.ts [hunk 1]
```diff
@@ -104,6 +101,23 @@ export class Popover extends SpectrumElement {
...
+                    this,
+                    `<${this.localName}> no longer supports the "dialog" attribute. Please slot an <sp-dialog> element into the <${this.localName}> instead.`,
+                    'https://opensource.adobe.com/spectrum-web-components/components/popover/#dialog-popovers',
+                    {
+                        type: 'api',
+                        level: 'deprecation',
+                    }
+                );
+            }
+        }
```
Comment (hunterloftis) [inline packages/popover/src/Popover.ts:117]: 🔥

**Review by hunterloftis — APPROVED** [review]

**Comment by Westbrook** [comment]
> Just making sure that these diffs are expected:
> 
> https://6fb3035dee9ee944fc231e1d21676146--spectrum-web-components.netlify.app/review/#OverlayStories/customizedClickContent.png: image
> 
> https://6fb3035dee9ee944fc231e1d21676146--spectrum-web-components.netlify.app/review/#OverlayStories/definedOverlayElement.png: image

We _could_ tweak the tests a little, to get them _closer_ to what they were, but I could this as expected for the updates made here:
1. Move from `<sp-popover dialog>` to `<sp-popover><sp-dialog>` increases the padding.
2. The use of `<sp-dialog size="s">` increases the `min-width`.

**Pull request merged**